File: ruby/backstage_item_updater.rb
class BackstageItemUpdater < BaseItemUpdater
  private

  def update_quality
    multiplier = case @item.sell_in
                 when 0
                   drop_quality
                   return
                 when 1..5 then 3
                 when 6..10 then 2
                 else 1
                 end
    safely_increase_quality(multiplier)
  end
end

File: js/src/backstage_item_updater.js
import BaseItemUpdater from "./base_item_updater";

export default class BackstageItemUpdater extends BaseItemUpdater {
  updateQuality() {
    let multiplier;
    switch (true) {
      case this.item.sellIn > 10:
        multiplier = 1;
        break;
      case this.item.sellIn > 5:
        multiplier = 2;
        break;
      case this.item.sellIn > 0:
        multiplier = 3;
        break;
      case this.item.sellIn === 0:
        this.dropQuality();
        return;
    }
    this.safelyIncreaseQuality(multiplier);
  }
}
